Rubberduck.Refactorings/EncapsulateField/EncapsulateFieldInsertNewCode/NewContentAggregator.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Refactorings.EncapsulateField
+{
+    public enum NewContentType
+    {
+        UserDefinedTypeDeclaration,
+        DeclarationBlock,
+        CodeSectionBlock,
+    }
+
+    public interface INewContentAggregator
+    {
+        /// <summary>
+        /// Allows gouping content blocks by <c>NewContentType</c>.
+        /// </summary>
+        void AddNewContent(NewContentType contentType, string block);
+        /// <summary>
+        /// Allows gouping content blocks by an adhoc identifier.
+        /// </summary>
+        void AddNewContent(string contentIdentifier, string block);
+        /// <summary>
+        /// Retrieves a block of content aggregated by <c>NewContentType</c>.
+        /// </summary>
+        /// <param name="newContentTypes"><c>NewContentType</c> blocks to aggregate</param>
+        string RetrieveBlock(params NewContentType[] newContentTypes);
+        /// <summary>
+        /// Retrieves a block of content aggregated by a user-determined identifier.
+        /// </summary>
+        /// <param name="contentIdentifiers"><c>NewContentType</c> blocks to aggregate</param>
+        string RetrieveBlock(params string[] contentIdentifiers);
+        /// <summary>
+        /// Sets default number of NewLines between blocks of code after
+        /// all retrieving block(s) of code.  The default value is 2.
+        /// </summary>
+        int NewLineLimit { set; get; }
+    }
+
+    /// <summary>
+    /// NewContentAggregator provides a repository for caching generated code blocks
+    /// and retrieving them as an aggregated single block of code organized by <c>NewContentType</c>.
+    /// </summary>
+    public class NewContentAggregator : INewContentAggregator
+    {
+        private readonly static string _doubleSpace = $"{Environment.NewLine}{Environment.NewLine}";
```

Comment:
Not sure I understand the motivation behind making this a public static string.  Since `$"{Environment.NewLine}{Environment.NewLine}"` is such a small and inherently clear expression as-is, there seems to be little gain in terms of 're-use' to define this publicly somewhere else...Can you say a bit more about what you were considering?  

Perhaps introducing a `string` extension function like `<string>.AppendNewLines(int count)` would be an option.  For the use case above, `_doubleSpace = string.Empty.AppendNewLines(2);`  in the constructor works...Or, get rid of `_doubleSpace` altogether and use the extension method inline.  `PrependNewLines(int)` would then be convenient to have as well.

Also, since `Environment.NewLine` is a static property/function - I do not see a way to define it as a resource expression directly.

Reply:
The reason I suggested that way was because I have a feeling I've seen that usage somewhere and instead of having multiple definition of "add 2 blank lines", define it once for common use.

I didn't mean as a resource. You would add to [`ApplicationConstants`](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Resources/ApplicationConstants.cs), perhaps, or as a new class.